Xamarin.Forms.Core/DependencyService.cs

```diff
@@ -24,15 +25,18 @@ public static class DependencyService
 		{
 			Initialize();
 
-			Type targetType = typeof(T);
-
-			if (!DependencyImplementations.ContainsKey(targetType))
+			DependencyData dependencyImplementation;
+			lock (Locker)
```

Comment:
Instead of creating a new type we can probably just lock on *DependencyImplementations* since it's private

Precedence
https://github.com/xamarin/Xamarin.Forms/blob/bd31e1e9fc8b2f9ad94cc99e0c7ab058174821f3/Xamarin.Forms.Core/Interactivity/AttachedCollection.cs#L64

Reply:
Locking on an object created for the purpose is clearer and easier for maintenance.

Reply:
That's fine (and I agree). I mainly suggested it because of how *SynchronizedList* and *AttachedCollection* lock on the internal lists so it was more a suggestion of consistency. That being said there are a couple places that use static objects which kills my argument.